util.c

```diff
@@ -33,6 +33,17 @@ char *xstrdup(const char *s)
 	return d;
 }
 
+char *xstrndup(const char *s, size_t n)
+{
+	size_t len = strnlen(s, n) + 1;
+	char *d = xmalloc(len);
+
+	memcpy(d, s, len - 1);
+	d[len - 1] = '\0';
+
+	return d;
+}
```

Comment:
Is there a compelling reason to do it this way, rather than just using `strndup()` and checking for a NULL return value?

Reply:
`strdup`/`strndup` are `_POSIX_C_SOURCE >= v200809L`. `xstrdup`, which already exists, isn't implemented in terms of `strdup`, so I followed suit for `xstrndup` as well.

---

Note: There is one `strdup` use in `tests/setprop_inplace.c`, everything else is xstrdup. I submitted #63 just now to address this.

Reply:
Ok, makes sense.